src/lazy/binary/raw/v1_1/type_descriptor.rs

```diff
@@ -176,39 +256,21 @@ pub struct Header {
     // The only time the `ion_type_code` is required is to distinguish between positive
     // and negative integers.
     pub ion_type_code: OpcodeType,
-    pub low_nibble: u8,
+    pub length_type: LengthType,
+    pub byte: u8,
 }
 
 impl Header {
+    pub fn byte(&self) -> u8 {
+        self.byte
+    }
+
+    pub fn low_nibble(&self) -> u8 {
+        self.byte & 0x0F
+    }
+
     pub fn length_type(&self) -> LengthType {
-        use LengthType::*;
-        match (self.ion_type_code, self.low_nibble) {
-            (OpcodeType::Boolean, 0xE..=0xF) => InOpcode(0),
-            (OpcodeType::Float, 0xA) => InOpcode(0),
-            (OpcodeType::Float, 0xB..=0xD) => InOpcode(1 << (self.low_nibble - 0xA)),
-            (OpcodeType::Integer, n) => InOpcode(n),
-            (OpcodeType::Nop, 0xC) => InOpcode(0),
-            (OpcodeType::NullNull, 0xA) => InOpcode(0),
-            (OpcodeType::String, 0..=15) => InOpcode(self.low_nibble),
-            (OpcodeType::InlineSymbol, n) if n < 16 => InOpcode(n),
-            (OpcodeType::SymbolAddress, n) if n < 4 => InOpcode(n),
-            (OpcodeType::Decimal, 0..=15) => InOpcode(self.low_nibble),
-            (OpcodeType::List, n) if n < 16 => InOpcode(n),
-            (OpcodeType::SExpression, n) if n < 16 => InOpcode(n),
-            (OpcodeType::TimestampShort, 0..=12) => {
-                InOpcode(ION_1_1_TIMESTAMP_SHORT_SIZE[self.low_nibble as usize])
-            }
-            (OpcodeType::TypedNull, _) => InOpcode(1),
-            (OpcodeType::Struct, n) if n < 16 => InOpcode(n),
-            (OpcodeType::DelimitedContainerClose, 0) => InOpcode(0),
-            (
-                OpcodeType::ListDelimited
-                | OpcodeType::SExpressionDelimited
-                | OpcodeType::StructDelimited,
-                _,
-            ) => Unknown,
-            _ => FlexUIntFollows,
-        }
+        self.length_type
```

Comment:
🗺️ This code block would often cause `BinaryBuffer::read_value_length` to not be inlined. Now this logic happens at compile time instead.